src/main/java/de/tum/in/www1/exerciseapp/service/StatisticService.java

```diff
@@ -159,18 +159,20 @@ public void updateStatisticsAndResults(QuizExercise quizExercise){
      * @param newResult the new Result, which will be added to the statistics
      * @param oldResult the old Result, which will be removedfrom the statistics. oldResult = null, if there is no old Result
      */
-    public boolean updateStatistics(Result newResult, Result oldResult) {
+    public boolean updateStatistics(Result newResult, Result oldResult, QuizExercise quiz) {
         // critical part locked with Semaphore statisticSemaphore
         try {
             statisticSemaphore.acquire();
 
-            QuizExercise quiz = quizExerciseRepository.findOne(newResult.getParticipation().getExercise().getId());
+            // get quiz and submission within semaphore to prevent lost updates
```

Comment:
This comment is too short. Please provide more details. What are "lost updates" 
I think this line is crucial so that the statistics work and we should be a bit more specific